PowerEditor/src/ScitillaComponent/Buffer.cpp

```diff
@@ -1426,6 +1426,10 @@ bool FileManager::loadFileData(Document doc, const TCHAR * filename, char* data,
 		success = false;
 	}
 
+	const int error = ferror(fp);
+	if (error != 0)
+		success = false;

```

Comment:
do this straight after the `fread()` call? as I expect that's what this is intended to check?

Reply:
agree. done.

Reply:
Obviously, it is up to @donho, but I believe the original solution was more efficient as well as more elegant..

Reply:
not sure what you mean by more efficient and more elegant 😕 

- there is a new `error` variable introduced in long code without limiting its scope
- the error checking is far removed from the actual call, which together with 'check global state' error handling seems like a real bad thing
- there is no guarantee that the `lenFile == 0` check would catch the error as the function might return less than the requested number of bytes before failing
- efficiency seems a non-issue as we're checking a variable as opposed to doing I/O which is an incomparable cost